src/OpenTelemetry.Instrumentation.MySqlData/MySqlDataInstrumentation.cs

```diff
@@ -40,15 +43,41 @@ public MySqlDataInstrumentation(MySqlDataInstrumentationOptions options = null)
         MySqlTrace.Switch.Level = SourceLevels.Information;
 
         // Mysql.Data removed `MySql.Data.MySqlClient.MySqlTrace.QueryAnalysisEnabled` since 8.0.31 so we need to set this using reflection.
-        var queryAnalysisEnabled = typeof(MySqlTrace).GetProperty("QueryAnalysisEnabled", BindingFlags.Public | BindingFlags.Static);
+        var queryAnalysisEnabled =
+            typeof(MySqlTrace).GetProperty("QueryAnalysisEnabled", BindingFlags.Public | BindingFlags.Static);
         if (queryAnalysisEnabled != null)
         {
             queryAnalysisEnabled.SetValue(null, true);
         }
+
+        // Mysql.Data add optional param `isAnalyzed` to MySqlConnectionStringBuilder constructor since 8.0.32
+        var ctor = typeof(MySqlConnectionStringBuilder).GetConstructor(new[] { typeof(string), typeof(bool) });
+        if (ctor == null)
+        {
+            ctor = typeof(MySqlConnectionStringBuilder).GetConstructor(new[] { typeof(string) })!;
```

Comment:
What is the expected behavior if this would be `null` as well?
Should we crash or just log the problem? Even if we crash we should better create a custom exception saying e.g. that the version of `MySql.Data` used by the application is not supported?

Reply:
Even if you agree it can be addressed as a separate PR.

Reply:
Added in [`47def4d` (#902)](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/902/commits/47def4db6274479bb458baf5fc15d25602501aef)